kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -298,6 +303,10 @@ private static Optional<DataSkippingPredicate> constructDataSkippingFilter(
     private static DataSkippingPredicate constructComparatorDataSkippingFilters(
             String comparator, Column leftCol, Literal rightLit, StatsSchemaHelper schemaHelper) {
 
+        boolean shouldAdjustCol = comparator.equals(">=") || comparator.equals(">");
+        Tuple2<Expression, Set<Column>> adjustedLeftCol = adjustTimestampForMaxColumnIfNeeded(
```

Comment:
I was honestly was thinking all this code would be under `statsHelper.getMaxColumn(...)` such that if it's a timestamp-type column and we're requesting a max column we do the addition there... I think this keeps this extra complexity hidden there and avoids complicating everything here. LMK what you think

Reply:
Yup, that was my thought as well but the need for an `Expression, Column` tuple made me apprehensive. But it's fewer corner cases to think through in the future as well (should we need more column expression adjustments). I'll update with a refactoring that accounts for this.